pkg/pod/entrypoint_lookup_impl_test.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pod
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/registry"
+	remotetest "github.com/tektoncd/pipeline/test"
+	"github.com/tektoncd/pipeline/test/parse"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakeclient "k8s.io/client-go/kubernetes/fake"
+)
+
+const (
+	username = "foo"
+	password = "bar"
+)
+
+type fakeHTTP struct {
+	reg http.Handler
+}
+
+func newfakeHTTP() fakeHTTP {
+	reg := registry.New()
+	return fakeHTTP{
+		reg: reg,
+	}
+}
+
+func (f *fakeHTTP) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Request authentication for ping request
+	// For further reference see https://docs.docker.com/registry/spec/api/#api-version-check
+	if r.URL.Path == "/v2/" && r.Method == http.MethodGet {
+		w.Header().Add("WWW-Authenticate", "basic")
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+	// Check auth if we've fetching the image
+	if strings.HasPrefix(r.URL.Path, "/v2/task") && r.Method == "GET" {
+		u, p, ok := r.BasicAuth()
+		if !ok || username != u || password != p {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+	}
+	// Default to open
+	f.reg.ServeHTTP(w, r)
+}
+
+func TestGet(t *testing.T) {
```

Comment:
nit: Rename this test case to be more descriptive - This isn't fully exhaustive in testing entrypoint.get - it's really only focused on the ImagePullSecrets.